app/services/estate_sso_mapper.rb

```diff
@@ -0,0 +1,56 @@
+class EstateSSOMapper
+  DIGITAL_ORG = 'digital.noms.moj'.freeze
+
+  def self.grouped_estates
+    @grouped_estates ||= begin
+      grouped_estates = Hash.new { |h, k| h[k] = [] }
+      Estate.where.not(group: nil).each do |estate|
+        grouped_estates[estate.group] << estate.sso_organisation_name
+      end
+      grouped_estates.values.freeze
+      grouped_estates.freeze
+    end
+  end
+
+  def self.reset_grouped_estates
```

Comment:
I'm not keen on mixing test and production code, this could easily be stubbed/mocked in tests